apps/next-evans/components/core/SearchInput.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import cx from 'classnames';
+import { MagnifyingGlassIcon } from '@heroicons/react/24/outline';
+
+interface Props {
+  query: string;
+  setQuery: (query: string) => any;
+  small?: boolean;
+  open?: boolean;
+  textColor?: string;
+  lineColor?: 'flgray-400' | 'flprimary';
+}
+
+const SearchInput: React.FC<Props> = ({
+  query,
+  setQuery,
+  small = false,
+  open = true,
+  lineColor = `flgray-400`,
+  textColor = `flgray-500`,
+}) => {
+  // purgeCSS: h-12 w-12 h-10 w-10 text-flgray-500
+  const size = small ? 10 : 12;
+  return (
+    <form
+      onSubmit={(e) => e.preventDefault()}
+      className={`rounded-full bg-white relative h-${size}`}
+    >
+      <input
+        type="text"
+        value={query}
+        onChange={(e) => setQuery(e.target.value)}
+        className={cx(
+          `SearchInput__input`,
+          // purgeCSS: border-flgray-400 border-flprimary
```

Comment:
these are places where i have to (or at least _had_ to - not 100% sure about newer tailwind) leave some hints so that tailwind didn't get confused by string interpolation. see my comment in the friend card area. in general i think it's usually better to pass around full tailwind classes so you don't have to do this kind of thing (instead of fragments + interpolation), but sometimes this is OK